files/image_config/platform/rc.local.j2

```diff
@@ -349,6 +349,8 @@ if [ -f $FIRST_BOOT_FILE ]; then
 
     systemctl enable snmp.timer
 
+    systemctl daemon-reexec
```

Comment:
'''
           Reexecute the systemd manager. This will serialize the manager
           state, reexecute the process and deserialize the state again. This
           command is of little use except for debugging and package upgrades.
           Sometimes, it might be helpful as a heavy-weight daemon-reload.
           While the daemon is being reexecuted, all sockets systemd listening
           on behalf of user configuration will stay accessible.
```

not sure what impact it has.

Reply:
can we do daemon-reload and then start the service one-by-one?

Reply:
which one is better?

Reply:
I believe starting them one by one results in a deadlock. When building a virtual switch image, check_install.py times out waiting for the first service to start. `daemon-reexec` is the only way I've found to start all services properly.

Reply:
`daemon-reexec` is cleaner and requires less maintenance (keeping the list of serivces up-to-date if we add/remove any).

That said, I'm not a fan of restarting all the services again right after boot. Seems hacky.

Reply:
I'm not sure there's another way. I've tried adjusting the dependencies for the services so that starting one of them results in all of them starting, but I still run into the timeout issue.

Reply:
what timeout issue? the other way to do it is to setup the service in initrd phase.

Reply:
I think this is a promising approach. can you investigate more on this approach?

idea of template unit files being used to create multiple instances of units.

https://fedoramagazine.org/systemd-template-unit-files/

Reply:
here is another link. https://www.digitalocean.com/community/tutorials/understanding-systemd-units-and-unit-files#creating-instance-units-from-template-unit-files

I think it makes sense to use namespace as instance so that we can build multiple instances of the same service.